android/src/main/java/io/ably/flutter/plugin/AblyInstanceStore.java

```diff
@@ -34,38 +34,100 @@
     private final LongSparseArray<AblyRest> restInstances = new LongSparseArray<>();
     private final LongSparseArray<AblyRealtime> realtimeInstances = new LongSparseArray<>();
     private final LongSparseArray<AsyncPaginatedResult<Object>> paginatedResults = new LongSparseArray<>();
+    private final AtomicLong nextHandle = new AtomicLong(1);
 
     static synchronized AblyInstanceStore getInstance() {
         return instance;
     }
 
     /**
-     * Returns a handle representing the next client that will be created. This handle can be used
-     * to get the client **after** it is instantiated using [createRest] or [createRealtime].
+     * A reserved client handle. Safe to be used from any thread.
+     *
+     * Instances support the creation of a single Rest or Realtime instance, where only one of the
+     * create methods may be called and it may only be called once.
      */
-    long getHandleForNextClient() {
-        return nextHandle;
+    interface ClientHandle {
```

Comment:
A little style suggestion - maybe the `ClientHandle` and/or it's implementation can be moved to a separate file? I'm not a fan of too long files, and from logical point of view, I'm not sure if it makes sense for this interface to be located inside `AblyInstanceStore`, since it's also used in other classes like `AblyMehtodCallHandler`. That's nothing blocking, since it's just related to the code style 🙂

Reply:
I am broadly supportive of that viewpoint, however there's always a balance in my view for such 'micro' interfaces where they only relate to an API offered by the primordial class in this file. The only reason `AblyMethodCallHandler` uses this interface is because it _also_ uses other code from this file for the same purpose.

So, in this case, I'm going to leave this as is. In future I would be open to it being refactored, if the context changes.